### PR TITLE
googleearth-pro: init at 7.3.3.7786

### PR DESCRIPTION
**Pull request by mofrim**
###### Motivation for this change

The old googleearth derivation maintained by @markus1189  was quite outdated and there have been some changes by google about how they distribute googleearth. The freely available version now is called "google-earth-pro" and there is no more i386 package avialable... as far as i did search. This is also the reason why i decided to make new package instead of updating the 'googleearth' package. This way the old version which is also available in i386 is still in nixpkgs but we can benefit from the most recent version of googleearth.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119095` at 4ddedd21 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>googleearth-pro</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/applications/misc/googleearth-pro/default.nix:93:3:
```
   |
93 |   checkPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/misc/googleearth-pro/default.nix:59:3:
```
   |
59 |   installPhase =''
   |   ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3391,6 +3391,12 @@
     githubId = 2129135;
     name = "Frederik Rietdijk";
   };
+  friedelino = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3394]: Please put this into a separate commit with the message ``maintainers: add friedelino`` and sort it alphabetically.

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  checkPhase = ''
+    $out/bin/gpsbabel -V > /dev/null
+  '';
+
+  dontPatchELF = true;
+
+  meta = with lib; {
+    description = "A world sphere viewer";
+    homepage = "http://earth.google.com";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/googleearth-pro/default.nix:101]:
```suggestion
    homepage = "https://earth.google.com";
```

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  src = fetchurl {
+    url = "https://dl.google.com/linux/earth/deb/pool/main/g/google-earth-pro-stable/google-earth-pro-stable_${version}-r0_${arch}.deb";
+    inherit sha256;
+  };
+
+  phases = [ "unpackPhase" "installPhase" "checkPhase" ];
+
+  doCheck = true;
+
+  buildInputs = [ dpkg ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/googleearth-pro/default.nix:52]:
```suggestion
  nativeBuildInputs = [ dpkg ];
```

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+in
+stdenv.mkDerivation rec {
+  pname = "googleearth-pro";
+  inherit version;
+  src = fetchurl {
+    url = "https://dl.google.com/linux/earth/deb/pool/main/g/google-earth-pro-stable/google-earth-pro-stable_${version}-r0_${arch}.deb";
+    inherit sha256;
+  };
+
+  phases = [ "unpackPhase" "installPhase" "checkPhase" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/googleearth-pro/default.nix:48]:
```suggestion
```

Please disable the phases which are causing issues when run.

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    libxml2
+    dbus
+    gst_all_1.gstreamer
+    gst_all_1.gst-plugins-base
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "googleearth-pro";
+  inherit version;
+  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/googleearth-pro/default.nix:43]:
```suggestion
  version = "7.3.3.7786";       # 7.1.8.3036 -> 7.3.3.7786

  src = fetchurl {
```

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    url = "https://dl.google.com/linux/earth/deb/pool/main/g/google-earth-pro-stable/google-earth-pro-stable_${version}-r0_${arch}.deb";
+    inherit sha256;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/googleearth-pro/default.nix:45]:
```suggestion
    sha256 = "1s3cakwrgf702g33rh8qs657d8bl68wgg8k89rksgvswwpd2zbb3";
```

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 7]
```diff
@@ -0,0 +1,106 @@
+{ lib, stdenv, fetchurl, glibc, libGLU, libGL, freetype, glib, libSM, libICE, libXi, libXv
+, libXrender, libXrandr, libXfixes, libXcursor, libXinerama, libXext, libX11, libXcomposite
+, libxcb, sqlite, zlib, fontconfig, dpkg, libproxy, libxml2, gst_all_1, dbus }:
+
+let
+  arch =
+    if stdenv.hostPlatform.system == "x86_64-linux" then "amd64"
+    else throw "Unsupported system ${stdenv.hostPlatform.system} ";
+  sha256 = "1s3cakwrgf702g33rh8qs657d8bl68wgg8k89rksgvswwpd2zbb3";
+  version = "7.3.3.7786";       # 7.1.8.3036 -> 7.3.3.7786
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/googleearth-pro/default.nix:10]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/googleearth-pro/default.nix (hunk 4 above)
Comment (friedelino) [reply pkgs/applications/misc/googleearth-pro/default.nix:47]: Hello Sandro First of all, thx for your review! Unfortunately I don't understand what you mean by disabling the phases causing issues when run. AFAIK package builds successfully. Do i need to perform additional tests?

Comment (SuperSandro2000) [reply pkgs/applications/misc/googleearth-pro/default.nix:47]: Overwriting phases is not allowed in leaf packages. If some phase is causing something to break disable it with ``dontBuild = true;``.

File: pkgs/applications/misc/googleearth-pro/default.nix [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    $out/bin/gpsbabel -V > /dev/null
+  '';
+
+  dontPatchELF = true;
+
+  meta = with lib; {
+    description = "A world sphere viewer";
+    homepage = "https://earth.google.com";
+    license = licenses.unfree;
+    maintainers = with maintainers; [ friedelino ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/googleearth-pro/default.nix:102]: You need to add yourself as a maintainer as part of this PR.

**Comment by SuperSandro2000** [comment]
Please squash all the package related commits together.

**Comment by friedelino** [comment]
> Please squash all the package related commits together.

Ok... i squashed all but the maintainer commits together. Then i force pushed to my branch. hope that's ok. now ofborg is complaining about me missing as a maintainer. is there something wrong about the order of my commits now?

**Comment by SuperSandro2000** [comment]
> is there something wrong about the order of my commits now?

That does not matter

@ofborg eval

File: maintainers/maintainer-list.nix [hunk 9]
```diff
@@ -3391,12 +3391,6 @@
     githubId = 2129135;
     name = "Frederik Rietdijk";
   };
-  friedelino = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix]: Why are you reverting it here?

Comment (friedelino) [reply maintainers/maintainer-list.nix]: argh! alright. because i in the beginning i did not have the editing of maintainers file in a seperate commit. i'll try to fix this...
